### PR TITLE
Upgrade dependencies and nightly

### PR DESCRIPTION
**Pull request by aschmois**
Story details: https://app.clubhouse.io/itprotv/story/67615

Changes:
- Upgraded nightly to 2021-03-19
- Use persistent static versions
- Added odd-jobs transitive dependency notes
- Update recurly to be compatible with nightly
- Remove unused `csv-conduit` package

### CODE REVIEW

File: snapshot.yaml [hunk 1]
```diff
@@ -15,15 +13,8 @@ packages:
...
-  # https://github.com/yesodweb/persistent/pull/1209
-  - git: https://github.com/yesodweb/persistent
-    commit: ec5f9ff9373f511d520bc8dd33cfab09db50cab9
-    subdirs:
-     - persistent
-     - persistent-mongoDB
-     - persistent-postgresql
-     - persistent-template
+  - persistent-2.12.0.0
+  - persistent-mongoDB-2.11.0.0
```
Comment (tfausak) [inline snapshot.yaml:17]: Is it alright to use `persistent` 2.12 with `persistent-mongoDB` 2.11? Also what happened to `persistent-postgresql` and `persistent-template`?

Comment (aschmois) [reply snapshot.yaml:17]: I wasn't 100% sure on this one, I assume the fixes were done on https://hackage.haskell.org/package/persistent-2.12.0.0 but there is no release for the other ones. `persistent-postgresql` and `persistent-template` are included in the nightly already (at least the latest matching on hackage).

Comment (aschmois) [reply snapshot.yaml:17]: nvm looks like 2.12 was a mistake https://github.com/yesodweb/persistent/pull/1211

Comment (codygman) [reply snapshot.yaml:17]: > Is it alright to use persistent 2.12 with persistent-mongoDB 2.11? Also what happened to persistent-postgresql and persistent-template?

The patch was in the `persistent` package and it's the only one that has a release like @aschmois said, but it does feel weird to not have the versions matching.

Comment (aschmois) [reply snapshot.yaml]: Looks like everything's already in stack's nightly, with the exception of mongo which has never landed in stackage for whatever reason.

**Comment by tfausak** [comment]
Waiting on https://github.com/EdutainmentLIVE/recurly/pull/3, right?

**Comment by aschmois** [comment]
> Waiting on [EdutainmentLIVE/recurly#3](https://github.com/EdutainmentLIVE/recurly/pull/3), right?

Yep, also waiting on a local build to see what can be removed

**Review by tfausak — APPROVED** [review]

**Pull request merged**